scipy/spatial/distance.py

```diff
@@ -130,12 +132,19 @@ def _copy_array_if_base_present(a):
     return a
 
 
-def _convert_to_bool(X):
-    return np.ascontiguousarray(X, dtype=bool)
+def _correlation_cdist_wrap(XA, XB, dm, **kwargs):
+    XA = XA - XA.mean(axis=1)[:, np.newaxis]
```

Comment:
Answering my own question: it was numpy 1.7 (https://docs.scipy.org/doc/numpy/release.html#numpy-1-7-0-release-notes), so using `keepdims` is fine.